src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private final int startLength = 10;
+    private T[] list = (T[]) (new Object[startLength]);
+    private int arraySize = 0;
 
     @Override
     public void add(T value) {
-
+        list[arraySize] = value;
+        arraySize++;
+        sizeControl();
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index <= size() && index >= 0) {
+            T temp = list[index];
+            list[index] = value;
+            for (int i = (index + 1); i < list.length; i++) {
+                value = list[i];
+                list[i] = temp;
+                temp = value;
+            }
+            arraySize++;
+            sizeControl();
+        } else {
+            indexError(index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexError(index);
+        return list[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexError(index);
+        list[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexError(index);
+        T removeValue = list[index];
+        for (int i = index; i < (list.length - 1); i++) {
+            list[i] = list[i + 1];
+        }
+        arraySize--;
+        return removeValue;
     }
 
     @Override
-    public T remove(T element) {
-        return null;
+    public T remove(T value) {
+        for (int i = 0; i < arraySize; i++) {
+            if (value != null && value.equals(list[i])) {
```

Comment:
```suggestion
            if (value == list[i] || value != null && value.equals(list[i])) {
```
так не потрібен другий if

Reply:
+